### PR TITLE
docs: Add go sample to clients

### PR DESCRIPTION
**Pull request by miniscruff**
# Pull request template

### Motivation and context

Add Go to the list of client examples

### Related issues

None, did on a whim from a slack request.

### Add tests to cover the changes

Just docs, not tests

### Sign your work

In order to contribute to the project, you must sign your work
https://github.com/scality/Guidelines/blob/master/CONTRIBUTING.md#sign-your-work

Thank you again for contributing! We will try to test and integrate the change
as soon as we can.

### CODE REVIEW

File: docs/CLIENTS.rst [hunk 1]
```diff
@@ -295,3 +295,60 @@ Should force path-style requests even though v3 advertises it does by default.
...
+      // more than the passed in timeout.
+      ctx := context.Background()
+      var cancelFn func()
+      if timeout > 0 {
+         ctx, cancelFn = context.WithTimeout(ctx, timeout)
+      }
+      // Ensure the context is canceled to prevent leaking.
+      // See context package for more information, https://golang.org/pkg/context/
+      if cancelFn != nil {
+         defer cancelFn()
```
Comment (rachedbenmustapha) [inline docs/CLIENTS.rst]: this can/should be in the same block as `cancelFn = context.WithTimeout`

File: docs/CLIENTS.rst [hunk 2]
```diff
@@ -295,3 +295,60 @@ Should force path-style requests even though v3 advertises it does by default.
...
+      os.Setenv("AWS_SECRET_ACCESS_KEY", "verySecretKey1")
+      endpoint := "http://localhost:8000"
+
+      timeout := time.Duration(10) * time.Second
+      sess := session.Must(session.NewSession())
+
+      // Create a context with a timeout that will abort the upload if it takes
+      // more than the passed in timeout.
+      ctx := context.Background()
+      var cancelFn func()
```
Comment (rachedbenmustapha) [inline docs/CLIENTS.rst]: - This doesn't need to be defined outside of the `if timeout > 0` block
- The usual name for this is just `cancel`

Comment (miniscruff) [reply docs/CLIENTS.rst]: Oh I also realized I hardcoded timeout anyway, something the example uses a flag for. Adjusted accordingly. ( https://github.com/aws/aws-sdk-go#complete-sdk-example is where I started from and includes both your comments lol )

**Review by naren-rajendran — APPROVED** [review]

**Review by jonathan-gramain — APPROVED** [review]

File: docs/CLIENTS.rst [hunk 3]
```diff
@@ -295,3 +295,50 @@ Should force path-style requests even though v3 advertises it does by default.
...
+   func main() {
+      os.Setenv("AWS_ACCESS_KEY_ID", "accessKey1")
+      os.Setenv("AWS_SECRET_ACCESS_KEY", "verySecretKey1")
+      endpoint := "http://localhost:8000"
+      timeout := time.Duration(10) * time.Second
+      sess := session.Must(session.NewSession())
+
+      // Create a context with a timeout that will abort the upload if it takes
+      // more than the passed in timeout.
+      ctx, cancel := context.WithTimeout(ctx, timeout)
```
Comment (rachedbenmustapha) [inline docs/CLIENTS.rst]: `ctx` is never assigned before here.

File: docs/CLIENTS.rst [hunk 4]
```diff
@@ -295,3 +295,50 @@ Should force path-style requests even though v3 advertises it does by default.
...
+      defer cancel()
+
+      svc := s3.New(sess, &aws.Config{
+         Region:   aws.String(endpoints.UsEast1RegionID),
+         Endpoint: &endpoint,
+      })
+
+      out, err := svc.ListBucketsWithContext(ctx, &s3.ListBucketsInput{})
+      if err != nil {
+         fmt.Printf("%s", err)
```
Comment (rachedbenmustapha) [inline docs/CLIENTS.rst]: `log.Fatal(err)` would be sort of more idiomatic.

**Review by rachedbenmustapha — CHANGES_REQUESTED** [review]

**Review by rachedbenmustapha — APPROVED** [review]

**Pull request merged**